src/v/cloud_storage/tests/s3_imposter.cc

```diff
@@ -36,6 +36,96 @@ inline ss::logger fixt_log("fixture"); // NOLINT
 /// For http_imposter to run this binary with a unique port
 uint16_t unit_test_httpd_port_number() { return 4442; }
 
+namespace {
+
+// Takes the input map of keys to expectations and returns a stringified XML
+// corresponding to the appropriate S3 response.
+ss::sstring list_objects_resp(
+  const std::map<ss::sstring, s3_imposter_fixture::expectation>& objects,
+  ss::sstring prefix,
+  ss::sstring delimiter) {
+    std::map<ss::sstring, size_t> content_key_to_size;
+    std::set<ss::sstring> common_prefixes;
+    // Filter by prefix and group by the substring between the prefix and first
+    // delimiter.
+    for (const auto& [_, expectation] : objects) {
+        auto& key = expectation.url;
+        vlog(fixt_log.trace, "Comparing {} to prefix {}", key, prefix);
+        if (key.size() < prefix.size()) {
+            continue;
+        }
+        bool match = true;
+        for (size_t i = 0; i < prefix.size(); ++i) {
+            if (key[i] != prefix[i]) {
+                match = false;
+            }
+        }
```

Comment:
Would `bool match = key.compare(0, prefix.size(), prefix) == 0` be acceptable here?

Reply:
Done